### ISSUE TITLE
Test case crash on rain_of_filth_i283.txt case

### ISSUE THREAD

**Issue by Rolzad73** [WagicProject/wagic#920]
Initially was running Qt Console Linux builds locally to investigate Travis build random crash issue (#912).
On my local machine (Ubuntu 16.10, latest everything I think), The crash was the following:

Segmentation fault ... in __dynamic_cast () from /usr/lib/x86_64-linux-gnu/libstdc++.so.6
on this call (link shows whole line)

[dynamic_cast<AManaProducer*> (aa)](https://github.com/WagicProject/wagic/blob/da033eb28e17b1390bba7b2e0e66eda7114e54e8/projects/mtg/src/GameObserver.cpp#L719)

I found that the trigger to the call was the "eot" after the "rain of filth" step.  This was the only test in the entire suite that caused a crash.

It appears that a fix related to this card was commited quite a while ago (3be21618d5193eec4912d2e7f0da2e43ce214a84).  Many changes have happened the source since then.

The "aa" object above was never NULL in my investigation.

This bug is outside of my expertise.

**Comment by kevlahnota** [comment]
I did add that ability (for Reflecting Pool and the likes). I guess I fixed it...

**Comment by kevlahnota** [comment]
I think the issue still there if I replace the forest with reflecting pool. It doesn't crash because it doesn't call the procedure...

**Comment by kevlahnota** [comment]
Pls let me know if you still have any issue with this #924

**Comment by Rolzad73** [comment]
I haven't seen this crash again on local testing.  I even changed that test in rain...283.txt to "reflecting pool" trying to test, but logs were giving me trouble to save to disk (I dunno what that's all about).

But also noticed another crash after the end of ALL the tests that said ...~QNetworkConfiguration().... but to me maybe was caused by an unused QCoreApplication in the Console main.  I deleted that just to trigger another Travis CI build to see what happens.
